tanner/tests/test_api.py

```diff
@@ -0,0 +1,224 @@
+        self.handler = Api(self.redis_client)
+        self.expected_content = ["8fa6aa98-4283-4085-bfb9-a1cd3a9e56e4"]
+
+        async def test():
+            self.returned_content = await self.handler.return_snares()
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.returned_content, self.expected_content)
+
+    def test_return_snares_error(self):
+        self.redis_client.smembers = AsyncMock(side_effect=aioredis.ProtocolError)
+        self.handler = Api(self.redis_client)
+
+        async def test():
+            self.returned_content = await self.handler.return_snares()
+
+        with self.assertLogs(level='ERROR') as log:
+            self.loop.run_until_complete(test())
+            self.assertIn('Can not connect to redis', log.output[0])
+
+    def test_return_snare_stats(self):
+        self.handler = Api(self.redis_client)
+        sessions = {
+            "sess1": {
+                "end_time": 2.00,
+                "start_time": 0.00,
+                "attack_types": ["lfi", "xss"]
+            },
+            "sess2": {
+                "end_time": 4.00,
+                "start_time": 2.00,
+                "attack_types": ["rfi", "lfi", "cmd_exec"]
+            }
+        }
+        self.handler.return_snare_info = AsyncMock(return_value=sessions)
+
+        self.expected_content = {
+            "attack_frequency": {
+                'cmd_exec': 1, 'lfi': 2, 'rfi': 1, 'sqli': 0, 'xss': 1
+            },
+            'total_duration': 4.0, 'total_sessions': 2
+        }
+
+        async def test():
+            self.returned_content = await self.handler.return_snare_stats(self.snare_uuid)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.returned_content, self.expected_content)
+
+    def test_return_snare_info(self):
+        query_res = ['{"sess1": {"end_time": 2.00, "start_time": 0.00 }, "sess2": {"attack_types": ["rfi"]}}']
+
+        self.redis_client.zrevrangebyscore = AsyncMock(return_value=query_res)
+        self.handler = Api(self.redis_client)
+        self.expected_content = [{'sess1': {'end_time': 2.0, 'start_time': 0.0}, 'sess2': {'attack_types': ['rfi']}}]
+
+        async def test():
+            self.returned_content = await self.handler.return_snare_info(self.uuid)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.expected_content, self.returned_content)
+
+    def test_return_snare_info_error(self):
+        self.redis_client.zrevrangebyscore = AsyncMock(side_effect=aioredis.ProtocolError)
+        self.handler = Api(self.redis_client)
+
+        async def test():
+            self.returned_content = await self.handler.return_snare_info(self.uuid)
+
+        with self.assertLogs(level='ERROR') as log:
+            self.loop.run_until_complete(test())
+            self.assertIn('Can not connect to redis', log.output[0])
+
+    def test_return_session_info(self):
```

Comment:
Add a case when `snare_uuid` is `None`. Make `return_snares` to return multiple snares id and make a `return_snare_info` method which returns different sessions for different values for snares ids.